Automattic/newspack-newsletters
# Add self-serve ads

### All Submissions:






### Changes proposed in this Pull Request:



Adds self-serve ads. Note that the ads will be placed at the bottom of the email – setting placement will be handled in another PR (issue: #237 ), as this one is pretty big already.

Closes #166 

### How to test the changes in this Pull Request:

1. Observe a new item under "Newsletters" menu in admin - "Ads". Visit that page:

![image](https://user-images.githubusercontent.com/7383192/85433035-20debc00-b584-11ea-86d7-be1b3792afaf.png)

2. Click "Create new ad". Create the content and try to publish – observe that it's not possible until an expiry date is set in the sidebar. If an expiry date is set in the past, a warning notice should be shown in the pre-publish panel.
3. Go back to the "Ads" view – observe the new ad being listed there. 
4. Create a new newsletter and send a test email. Observe the ad being appended to the bottom of the email (but above the ESP footer)
5. Send the newsletter – observe a notice in the pre-send modal about the ads which will be inserted. 
6. Observe the ad being placed in the email, just like in the test email.
7. Create a couple of ads which are drafts, published, expired – to test the UI in "Ads" view

![image](https://user-images.githubusercontent.com/7383192/85434022-aca51800-b585-11ea-83ba-147db1761632.png)


### Other information:
